controllers/user/UserController.php

```diff
@@ -35,6 +35,12 @@ public function actionCreate(){
             $model = new RegistrationChecks();
             $legalKey = $model->load(Yii::$app->request->getBodyParam("legal", []), '');
 
+            //Only allow 1, true, false and 0 value of dataPrivacyCheck key
+            $checkKeyValue = array(1,true,false,0);
+            if(!in_array($model->dataPrivacyCheck,$checkKeyValue,true)){
+                return $this->returnError(400,"Invalid value of dataPrivacyCheck key");
+            }
```

Comment:
shouldn't this be checked, after the `legal` key is checked? if the `legal` key is not present a `dataPrivacyCheck` is also missing and code does not to be check `1`, `true`, `false` and `0`, right? can you switch the order please and then i think we made it 🥳